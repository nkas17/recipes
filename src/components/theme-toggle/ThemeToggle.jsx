import React from 'react';

class ThemeToggle extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      theme: 'light',
    };
    this.toggleTheme = this.toggleTheme.bind(this);
  }

  toggleTheme() {
    const { theme } = this.state;
    if (theme === 'light') {
      this.setState({ theme: 'dark' });
      const app = document.getElementById('app');
      app.className = 'dark';
    } else {
      this.setState({ theme: 'light' });
      const app = document.getElementById('app');
      app.className = 'light';
    }
  }

  render() {
    const { theme } = this.state;
    return (
      <button
        type="button"
        className="btn btn-link theme-toggle"
        onClick={this.toggleTheme}
      >
        {theme}
      </button>
    );
  }
}

export default ThemeToggle;
