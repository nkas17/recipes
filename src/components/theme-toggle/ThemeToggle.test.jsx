import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ThemeToggle from './ThemeToggle';

describe('ThemeToggle', () => {
  it('renders correctly', () => {
    const tree = shallow(<ThemeToggle />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('runs toggleTheme correctly when theme is light', () => {
    const themeToggle = new ThemeToggle();
    themeToggle.setState = jest.fn();
    const app = {};
    document.getElementById = jest.fn().mockReturnValue(app);
    themeToggle.toggleTheme();
    expect(themeToggle.setState).toHaveBeenCalledWith({ theme: 'dark' });
    expect(document.getElementById).toHaveBeenCalledWith('app');
    expect(app.className).toBe('dark');
  });

  it('runs toggleTheme correctly when theme is dark', () => {
    const themeToggle = new ThemeToggle();
    themeToggle.state = { theme: 'dark' };
    themeToggle.setState = jest.fn();
    const app = {};
    document.getElementById = jest.fn().mockReturnValue(app);
    themeToggle.toggleTheme();
    expect(themeToggle.setState).toHaveBeenCalledWith({ theme: 'light' });
    expect(document.getElementById).toHaveBeenCalledWith('app');
    expect(app.className).toBe('light');
  });
});
