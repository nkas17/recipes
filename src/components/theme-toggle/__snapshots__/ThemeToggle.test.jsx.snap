// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle renders correctly 1`] = `
<button
  className="btn btn-link theme-toggle"
  onClick={[Function]}
  type="button"
>
  light
</button>
`;
